### ISSUE TITLE
Running elm make in parallell causes corrupt dependency

### ISSUE THREAD

**Issue by smucode** [elm/compiler#1830]
From time to time I noticed this in our CI.

```
$ elm make --output=/dev/null src/elm/Main.elm
Dependencies loaded from local cache.
Verifying dependencies...
Building dependencies (1/47)
(snip...)
Building dependencies (47/47)
Dependencies ready!                
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

lerna ERR! yarn run build:elm stderr:
-- CORRUPT DEPENDENCY ----------------------------------------------------------

I ran into a problem while building the following package:

    elm/url 1.0.0

This probably means the downloaded files got corrupted somehow. Try deleting
/root/go/src/repo/main/frontend/elm-home (a directory for
caching build artifacts) and see if that resolves the issue.

error Command failed with exit code 1.
```

We are using lerna to build two projects in parallell. Building them sequentially works fine.

SSCCE: https://github.com/smucode/elm-make-parallell

**Comment by evancz** [comment]
Is this on two separate projects that are both downloading dependencies at the same time?

Is this something you can do with other compilers that have a per-user cache of dependencies? If so, which languages? How do they make some sort of reliable "file system lock" so that a dependency is never downloaded by two processes who need the same resource and do not know about each other?

**Comment by smucode** [comment]
Yes, we were building two separate projects that were both downloading dependencies at the same time. We only experienced this in CI, and was easily fixed by not building in parallell.

**Comment by evancz** [comment]
I added a filelock when building dependencies in https://github.com/elm/compiler/commit/054fa04074fd819958affd04a526e10f69aa5197 which should make it impossible for two simultaneous `elm make` runs interfere with each others files.

Thank you for the report! This fix should become available when 0.19.1 comes out with a bunch of other improvements.

**Comment by domenkozar** [comment]
Note that this problem still exists with 0.19.1

```
Compiling ...+-------------------------------------------------------------------------------
|  Corrupt File: /build/source/myproject/elm-stuff/0.19.1/i.dat
|   Byte Offset: 102336
|       Message: not enough bytes
|
| Please report this to https://github.com/elm/compiler/issues
| Trying to continue anyway.
+-------------------------------------------------------------------------------

-- CORRUPT CACHE ---------------------------------------------------------------

It looks like some of the information cached in elm-stuff/ has been corrupted.

Try deleting your elm-stuff/ directory to get unstuck.

Note: This almost certainly means that a 3rd party tool (or editor plugin) is
causing problems your the elm-stuff/ directory. Try disabling 3rd party tools
one by one until you figure out which it is!
```
